src/equiv.js

```diff
@@ -212,8 +236,75 @@ QUnit.equiv = (function() {
 
 			// apply transition with (1..n) arguments
 		}( args[ 0 ], args[ 1 ] ) ) &&
-			innerEquiv.apply( this, args.splice( 1, args.length - 1 ) ) );
-	};
+			deepEquiv.apply( this, args.splice( 1, args.length - 1 ) ) );
+	},
+	propEquiv = function ( ) {
+		var args = [].slice.apply( arguments );
+		if ( args.length < 2 ) {
+			return true;
+		}
+
+		return ( (function ( a, b ) {
+			if ( !QUnit.is( "object", a ) || !QUnit.is( "object", b ) ) {
+				// if the test author submits parameters that are not objects, check for deepEquiv
+				return deepEquiv( a, b );
+			} else {
+				/*jshint forin:false */
+				var i, j, loop, aCircular, bCircular, currentProperty, subEquiv,
+				// Default to true
+				eq = true,
+				aProperties = ownObjectKeys( a ),
+				bProperties = ownObjectKeys( b );
+
+				// stack constructor before traversing properties
+				callers.push( a.constructor );
+
+				// track reference to avoid circular references
+				parents.push( a );
+				parentsB.push( b );
+
+				for ( i = 0; i < aProperties.length; i++ ){
+					currentProperty = aProperties[i];
+					loop = false;
+					for ( j = 0; j < parents.length; j++ ) {
+
+						aCircular = parents[j] === a[currentProperty];
+						bCircular = parentsB[j] === b[currentProperty];
+
+						if ( aCircular || bCircular ){
+							if ( a[ currentProperty ] === b[ currentProperty ] ||
+									aCircular && bCircular ){
+								loop = true;
+							} else {
+								eq = false;
+								break;
+							}
+						}
+
+					}
+
+					// we'll only use prop equivalence to compare objects
+					subEquiv = QUnit.is( "object", a[ currentProperty ] ) ?
+						propEquiv : deepEquiv;
+
+					if ( !loop && !subEquiv( a[ currentProperty ], b[ currentProperty ] ) ) {
+						eq = false;
+						break;
+					}
+				}
+				parents.pop();
+				parentsB.pop();
+				callers.pop(); // unstack, we are done
+
+				// ensure both objects have the same properties
+				return eq && deepEquiv(aProperties.sort(), bProperties.sort());
+			}
+		}( args[ 0 ], args[ 1 ] ) ) &&
+			propEquiv.apply(this, args.splice( 1, args.length - 1 ) ) );
+
+	},
+	innerEquiv = deepEquiv; // default equivalence check
```

Comment:
Might as well drop `innerEquiv` and assign `deepEquiv.props` directly, then return that.